From 6c9641888e29f925813d5a4c35e17b6dec0fbbae Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:04:27 -0400
Subject: [powerpc] pci: Don't use bitfield for force_32bit_msi

Message-id: <1410545655-205645-39-git-send-email-dzickus@redhat.com>
Patchwork-id: 94111
O-Subject: [RHEL7 PATCH 038/626] powerpc/pci: Don't use bitfield for force_32bit_msi
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 0c69f9c52c474ccd1cade27c7be5f8ab830cdc3a
Author: Anton Blanchard <anton@samba.org>
Date:   Wed Aug 7 02:01:25 2013 +1000

    powerpc/pci: Don't use bitfield for force_32bit_msi

    Fix a sparse warning about force_32bit_msi being a one bit bitfield.

    Signed-off-by: Anton Blanchard <anton@samba.org>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/include/asm/pci-bridge.h b/arch/powerpc/include/asm/pci-bridge.h
index 32d0d20..4ca90a3 100644
--- a/arch/powerpc/include/asm/pci-bridge.h
+++ b/arch/powerpc/include/asm/pci-bridge.h
@@ -159,7 +159,7 @@ struct pci_dn {
 
  int pci_ext_config_space; /* for pci devices */
 
- int force_32bit_msi:1;
+ bool force_32bit_msi;
 
  struct pci_dev *pcidev; /* back-pointer to the pci device */
 #ifdef CONFIG_EEH
diff --git a/arch/powerpc/kernel/pci_64.c b/arch/powerpc/kernel/pci_64.c
index 8911c87..155013d 100644
--- a/arch/powerpc/kernel/pci_64.c
+++ b/arch/powerpc/kernel/pci_64.c
@@ -272,7 +272,7 @@ static void quirk_radeon_32bit_msi(struct pci_dev *dev)
  struct pci_dn *pdn = pci_get_pdn(dev);
 
  if (pdn)
-  pdn->force_32bit_msi = 1;
+  pdn->force_32bit_msi = true;
 }
 DECLARE_PCI_FIXUP_FINAL(PCI_VENDOR_ID_ATI, 0x68f2, quirk_radeon_32bit_msi);
 DECLARE_PCI_FIXUP_FINAL(PCI_VENDOR_ID_ATI, 0xaa68, quirk_radeon_32bit_msi);
-- 
1.7.1